skills_ml/algorithms/skill_feature_creator/__init__.py

```diff
@@ -1,81 +1,121 @@
+from functools import reduce
+
+from skills_ml.algorithms.job_vectorizers.doc2vec_vectorizer import Doc2Vectorizer
+
 class FeatureCreator(object):
     """ Feature Creator Factory that help users to instantiate different
     types of feature at once and combine them together.
 
     Example:
-        doc = "some job posting..."
-        feature_vector = FeatureCreate(doc).combine(features="all")
-        feature_vector = FeatureCreate(doc).combine(features=["StructuralFeature", "EmbeddingFeature"])
+        from airflow.hooks import S3Hook
+        from skills_ml.algorithms.skill_feature_creator import FeatureCreator
+
+        s3_conn = S3Hook().get_conn()
+
+        docs=["something", "something2"]
+
+        feature_vector_generator = FeatureCreator(s3_conn, features="all").transform(docs)
+        feature_vector_generator = FeatureCreator(s3_conn, features=["StructuralFeature", "EmbeddingFeature"]).transform(docs)
 
     Args:
         doc (string): job posting data.
     """
-    def __init__(self, doc):
-        self.all_features = [ f.__name__ for f in FeatureCreator.__subclasses__()]
-        self.doc = doc
-
-    @staticmethod
-    def _factory(type, doc):
-        if type == "StructuralFeature":
-            return StructuralFeature(doc)
-
-        if type == "ContextualFeature":
-            return ContextualFeature(doc)
+    def __init__(
+            self,
+            s3_conn=None,
+            features="all",
+            embedding_model_name="gensim_doc2vec_va_0605",
```

Comment:
It would be nice to avoid these defaults being implemented in multipke places (here and in Doc2Vectorizer). Maybe make these defaults into Nones, and rely on Doc2Vectorizer to pick the default. The code in the latter would have to be changed, to make the default into None and then have the initialization say `self.model_name = model_name or 'gensim...`